app/Providers/AppServiceProvider.php

```diff
@@ -16,6 +19,8 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
+        $this->app->singleton(TransactionReader::class, fn (Application $app) => new PhpSpreadsheetAdapter());
```

Comment:
ℹ️ I'm using the [PhpSpreadsheet](https://github.com/PHPOffice/PhpSpreadsheet) to read the csv file, as it supports other formats out of the box that I'd like to support as well down the line. I've got to say the library isn't very intuitive though